bolt/include/bolt/Passes/IdenticalCodeFolding.h

```diff
@@ -27,15 +28,74 @@ class IdenticalCodeFolding : public BinaryFunctionPass {
       return false;
     if (BF.hasSDTMarker())
       return false;
+    if (BF.hasAddressTaken())
+      return false;
     return BinaryFunctionPass::shouldOptimize(BF);
   }
 
 public:
+  enum class ICFLevel {
+    None, // No ICF. (Default)
+    Safe, // Safe ICF for all sections.
+    All,  // Aggressive ICF for code.
```

Comment:
For automatic doc generation (doxygen), you can use "///". The last two comments are confusing as one says "for all sections" and the other - "for code".